lua/telescope/_extensions/file_browser/make_entry.lua

```diff
@@ -218,9 +229,11 @@ local make_entry = function(opts)
     return rawget(t, rawget({ value = 1 }, k))
   end
 
-  return function(absolute_path)
+  return function(entry)
+    local absolute_path = entry[1]
     local e = setmetatable({
       absolute_path,
+      git = entry[2],
       ordinal = (absolute_path == opts.cwd and ".")
         or (absolute_path == parent_dir and ".." or absolute_path:sub(cwd_substr, -1)),
     }, mt)
```

Comment:
This was a massive gotcha for me because the telescope docs say
```
It's best practice to have a value reference to the original entry, that way we will always have access to the complete table in our action.
```
(https://github.com/nvim-telescope/telescope.nvim/blob/master/developers.md#entry-maker)

which is not followed here but instead I assume the `__index` function is somehow overwriting this behaviour.

Reply:
> It's best practice to have a value reference to the original entry, that way we will always have access to the complete table in our action.

Mhm, `value` is the original entry (i.e. absolute path) as produced by whatever comes "into" the finder. At least is how I understand that convention.

The complete table in the example refers to the original entries in that context, i.e.

```lua
      results = {
        { "red", "#ff0000" },
        { "green", "#00ff00" },
        { "blue", "#0000ff" },
      }
```

I don't see why you think that's not followed here.

> the __index function is somehow overwriting this behaviour.

The __index metamethod primarily serves lazy attribute creation (e.g. `plenary.Path`, `stat`) and order for access. Strictly speaking we could allow users to pass their own display function if it was part of entry.

Reply:
That table going into the finder is in this case a list of all the found files as an absolute path in the cwd and their git status like this:
```lua
results = {
        { "/abs/path1", " M" },
        { "/abs/path2", "??" },
        { "/abs/path3", "  " },
      }
```

However, when trying to access `entry.value` from within the `display` function it is referring to the absolute path and not to `{ "/abs/path1", " M" }` what I would expect based on the description from the telescope docs.